### PR TITLE
Fix warnings in SVGA_S3_ReadSEQ

### PR DESCRIPTION
**Pull request by dreamer**
GCC warns us about these lines:

vga_s3.cpp:482:42: warning: converting the result of '<<' to a boolean;
did you mean '(vga.s3.mclk.r << 5) != 0'? [-Wint-in-bool-context]

In this case the answer is: no, we want to use bit-or instead of
logical-or instead.

Code in SVGA_S3_WriteSEQ is an I/O callback called with iolen == 1, and
value passed via val is originally uint8_t.

The 8-bit value is then split into 5-bit value stored in byte n, and
3-bit value stored in byte r.

Therefore we can conclude SVGA_S3_ReadSEQ is supposed to recombine those
two bytes back into a single value, and not merely return 1 if any bits
were set.

### CODE REVIEW

**Comment by dreamer** [comment]
@kcgen 
New Clang on macOS learned to detect [-Wint-in-bool-context] the same way GCC do, so our 0 warning limit on macOS is no longer valid. Instead of raising the limit I opted to finish the fix I had prepared a few days ago :)

Unfortunately, there is new packaging error on macOS and I don't understand what is causing it :( (glib 2.66.2 is missing… despite this version being exactly the file detected by pkgconfig…

@rderooy @Wengier You might be interested in this change as well - I noticed these lines in `vga_s3.cpp` are marked with FIXMEs in DOSBox-X source code.

I don't know any software that is affected by this - I made the change based compiler warning and analysis of surrounding code. This code is triggered when a program unlocks protected registers on S3 Trio card, writes to them, and then reads them back. I believe these registers are protected for backwards-compatibility with older graphics cards, but I'm not sure.

**Comment by Wengier** [comment]
@dreamer Thanks for letting me know the SVGA_S3 fixes! You are absolutely right that both case 0x10 and case 0x12 in the SVGA_S3_ReadSEQ function are marked with FIXMEs in DOSBox-X source code. I have already modified this code with your authorship (https://github.com/joncampbell123/dosbox-x/commit/53afe720e6edeed963e4ed8bce49ea70f1223ebe), commenting out the old code (https://github.com/joncampbell123/dosbox-x/commit/53afe720e6edeed963e4ed8bce49ea70f1223ebe). Thanks for the improvements, and I noticed that @joncampbell123 has just left a comment too (see https://github.com/CartoonFan/dosbox-x/commit/53afe720e6edeed963e4ed8bce49ea70f1223ebe).

P.S. I have not modified the SVGA_S3_WriteSEQ code there yet because I noticed that it may not build correctly in DOSBox-X because the code appears to require C++14 or later which is not yet enforced in DOSBox-X.

File: src/hardware/vga_s3.cpp [hunk 1]
```diff
@@ -446,15 +446,15 @@ void SVGA_S3_WriteSEQ(Bitu reg,Bitu val,Bitu iolen) {
 		vga.s3.pll.lock=val;
 		break;
 	case 0x10:		/* Memory PLL Data Low */
-		vga.s3.mclk.n=val & 0x1f;
-		vga.s3.mclk.r=val >> 5;
+		vga.s3.mclk.n = (val & 0b000'11111);
+		vga.s3.mclk.r = (val & 0b111'00000) >> 5;
```
Comment (kcgen) [inline src/hardware/vga_s3.cpp:450]: bit-notation is so immensely helpfully for me to see what bits are being passes through, as my mind doesn't automatically convert 0x1f to a bit-mask.

File: src/hardware/vga_s3.cpp [hunk 2]
```diff
@@ -479,11 +479,11 @@ Bitu SVGA_S3_ReadSEQ(Bitu reg,Bitu iolen) {
 	case 0x08:		/* PLL Unlock */
 		return vga.s3.pll.lock;
 	case 0x10:		/* Memory PLL Data Low */
-		return vga.s3.mclk.n || (vga.s3.mclk.r << 5);
+		return vga.s3.mclk.n | (vga.s3.mclk.r << 5);
 	case 0x11:		/* Memory PLL Data High */
 		return vga.s3.mclk.m;
 	case 0x12:		/* Video PLL Data Low */
-		return vga.s3.clk[3].n || (vga.s3.clk[3].r << 5);
+		return vga.s3.clk[3].n | (vga.s3.clk[3].r << 5);
```
Comment (kcgen) [inline src/hardware/vga_s3.cpp:486]: Can we flip the return fields so their order flows with the bits-fields? 0b111'00000 | 0b000'11111; (that's how I visually them being put back together into an 8-bit number).

`return (vga.s3.mclk.r << 5) | vga.s3.mclk.n;`

`return (vga.s3.clk[3].r << 5) | vga.s3.clk[3].n;`

(purely aesthetic :-)

**Review by kcgen — APPROVED** [review]
Only one minor comment; very nice fix and readability improvement.
Hopefully picked up by upstream too.

**Comment by dreamer** [comment]
Thanks for the feedback, everyone :)

**Pull request merged**